xlators/cluster/afr/src/afr-self-heald.c

```diff
@@ -704,9 +701,8 @@ afr_shd_ta_unset_xattrs(xlator_t *this, loc_t *loc, dict_t **xdata, int healer)
             goto out;
         }
 
-        memcpy(pending, pending_raw, sizeof(pending));
         for (j = 0; j < AFR_NUM_CHANGE_LOGS; j++) {
-            val = ntoh32(pending[j]);
+            val = ntoh32(*((int *)pending_raw + j));
```

Comment:
What is the need of it? we can just check *((int *)pending_raw + j) in "if" and use it in 718

Reply:
@aspandey If we don't use a temporary variable (val), the expression will be evaluated at 2 places, in the"if" and the "hton32" at 718 right (unless it gets optimized)?

Reply:
However, we do not need the conversion here - it's only relevant if val is non-zero, which we can check without ntoh32.
And if we refrain from the conversion, we do not need to convert it back later on?

Reply:
True. It is becoming a bit unsightly but If we are making this optimization better go all the way.
```

@@ -702,8 +701,7 @@ afr_shd_ta_unset_xattrs(xlator_t *this, loc_t *loc, dict_t **xdata, int healer)
         }

         for (j = 0; j < AFR_NUM_CHANGE_LOGS; j++) {
-            val = ntoh32(*((int *)pending_raw + j));
-            if (val) {
+            if (*((int *)pending_raw + j)) {
                 if (i == healer) {
                     gf_msg(this->name, GF_LOG_INFO, 0, AFR_MSG_THIN_ARB,
                            "I am "
@@ -715,7 +713,7 @@ afr_shd_ta_unset_xattrs(xlator_t *this, loc_t *loc, dict_t **xdata, int healer)
                     goto out;
                 }
                 need_xattrop = _gf_true;
-                raw[j] = hton32(-val);
+                raw[j] = -(*((int *)pending_raw + j))
             }
         }
```

Reply:
I think I will still use 'val' but skip the conversion.